packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -484,6 +468,15 @@ export class EdrProviderWrapper
     }
   }
 
+  /**
+   * Injects a `VMTracer` that observes EVM throughout requests.
+   *
+   * Used for internal stack traces integration tests.
+   */
+  public injectVmTracer(vmTracer?: VMTracer) {
```

Comment:
Nit: I'm not a fan of the use of the word "inject" here. It seems to be used both to set and to reset (to `undefined`), so I'd just keep it straightforward with:
```suggestion
  public setVmTracer(vmTracer?: VMTracer) {
```

Reply:
Sure, makes sense - I didn't want to use `set` myself because I didn't want to read it like it's a regular a config setting but rather a special mode, however, I understand the set/unset argument here :+1: 

Changed to `setVmTracer` in https://github.com/NomicFoundation/hardhat/pull/5429/commits/2fe2cff7863bd69c22e2b7669fdf0a88f5c3d77c